### PR TITLE
fix(websocket): use binary operations instead of a Set for closeStatus

### PR DESCRIPTION
**Pull request by Uzlopak**
Follow up to #3628

### CODE REVIEW

**Review by KhafraDev — CHANGES_REQUESTED** [review]
> It's possible to avoid the set *but nothing here is a hot path - websockets are not closed often.*

It's not worth the mental overhead. The websocket closing handshake is an extremely cold path.

**Comment by Uzlopak** [comment]
Using a Set is an unconventional and to honest very ugly Implementation in our codebase. I strongly recommend to avoid ES6 Sets to just store 2 bit information...

File: lib/web/websocket/websocket.js [hunk 1]
```diff
@@ -671,9 +671,7 @@ class WebSocket extends EventTarget {
     // If the TCP connection was closed after the
     // WebSocket closing handshake was completed, the WebSocket connection
     // is said to have been closed _cleanly_.
-    const wasClean =
-      this.#handler.closeState.has(sentCloseFrameState.SENT) &&
-      this.#handler.closeState.has(sentCloseFrameState.RECEIVED)
+    const wasClean = this.#handler.closeState === (sentCloseFrameState.SENT & sentCloseFrameState.RECEIVED)
```
Comment (Uzlopak) [inline lib/web/websocket/websocket.js:674]:
```suggestion
    const wasClean = this.#handler.closeState === (sentCloseFrameState.SENT | sentCloseFrameState.RECEIVED)
```

**Comment by KhafraDev** [comment]
I don't agree with any of that. The implementation is easy to understand and not in a critical path. Those paths are triggered when
- manually closing (`WebSocket.prototype.close`)
- receiving a close frame

You can only close a websocket once per instance, and websockets are rarely closed by nature. Not only that, `NOT_SENT` is an illogical name for a state because the closing handshake starts when sending **or** receiving a close frame.

**Comment by Uzlopak** [comment]
@KhafraDev 

What if rename it to NONE?

**Pull request closed without merging**